### ISSUE TITLE
Selecting text from debug console history view is really hard

### ISSUE THREAD

**Issue by rebornix** [microsoft/vscode#130083]
Issue Type: <b>Bug</b>

* Can't select from the beginning of a line
* Selecting from the middle of a line works, but really hard to select to the beginning of the line

VS Code version: Code - Insiders 1.59.0-insider (cead2f699cd0afb2464d8d3ad975776051583b4f, 2021-08-03T16:24:47.105Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 x 3600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.93GB (7.97GB free)|
|Process Argv|--enable-proposed-api joyceerhl.vscode-pyolite --enable-proposed-api rebornix.better-jupyter-errors --enable-proposed-api donjayamanne.kusto --crash-reporter-id e73ca734-e67c-473d-8c47-80c5407fc859|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by isidorn** [comment]
You are right. The reason for this is that we are using a tree underneath, and because of that, on the left side there is space left for the twistie which is a separate html element. Clicking on that twistie space does not allow you to native select the other HTML element.
I think we could improve this with some css tweaks to make the content element having some padding left and by reducing the twistie element size.
For now adding a help-wanted so we see if somebody from the community would like to tackle this.
Maybe `hideTwistieOfChildlessElements` tree flag could help

Code poitner https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/repl.ts#L558

**Comment by bpceee** [comment]
Hi @isidorn 
I just created a PR for this. 
It doesn't use the `hideTwistieOfChildlessElements` flag as we also need to handle twisties with children.

**Comment by isidorn** [comment]
@bpceee thanks for the PR. I have commented in the PR, so let's continue the discussion there.

**Comment by mojo2012** [comment]
Selecting from the left side of the view is not the only problem. Selecting text beyond the visible viewport is near to impossible:
* It doesn't scroll automatically when you are reaching the top or bottom border. 
* Furthermore the selected text is lost if you scroll to far. 
* Sometimes by scrolling up and down the selection is lost or changed even

**Comment by roblourens** [comment]
Thanks for the report — this has been addressed. Commit https://github.com/microsoft/vscode/commit/dd86be1a954 added proper user text-selection support to list/tree views and adopted it in the debug console REPL. This handles selecting from the beginning of lines, auto-scrolling near viewport edges, and preserving the selection while scrolling. It shipped in VS Code 1.100. Please open a new issue if you still see selection problems on the latest build.
